# This patch file was generated by NetBeans IDE
# Following Index: paths are relative to: /space/hudson/hudson/main/core
# This patch can be applied using context Tools: Patch action on respective folder.
# It uses platform neutral UTF-8 encoding and \n newlines.
# Above lines and this line are ignored by the patching process.
Index: src/main/java/hudson/model/Hudson.java
--- src/main/java/hudson/model/Hudson.java Base (BASE)
+++ src/main/java/hudson/model/Hudson.java Locally Modified (Based On LOCAL)
@@ -2124,7 +2124,8 @@
      * If the user chose the default JDK, make sure we got 'java' in PATH.
      */
     public void doDefaultJDKCheck( StaplerRequest req, StaplerResponse rsp ) throws IOException, ServletException {
-        new FormFieldValidator(req,rsp,true) {
+        //no need to check admin here
+        new FormFieldValidator(req,rsp,false) {
             public void check() throws IOException, ServletException {
                 String v = request.getParameter("value");
                 if(!v.equals("(Default)"))
@@ -2148,7 +2149,8 @@
     public void doItemExistsCheck(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
         // this method can be used to check if a file exists anywhere in the file system,
         // so it should be protected.
-        new FormFieldValidator(req,rsp,true) {
+
+        new FormFieldValidator(req,rsp,false) {
             protected void check() throws IOException, ServletException {
                 String job = fixEmpty(request.getParameter("value"));
                 if(job==null) {
@@ -2290,6 +2292,7 @@
      * @throws ServletException thrown by FormFieldValidator.check()
      */
     public void doCheckURIEncoding(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
+        //not sure what exactly this does, but suppose again admin check is not necessary?!
         new FormFieldValidator(req, rsp, true) {
             @Override
             protected void check() throws IOException, ServletException {
Index: src/main/java/hudson/scm/SubversionSCM.java
--- src/main/java/hudson/scm/SubversionSCM.java Base (BASE)
+++ src/main/java/hudson/scm/SubversionSCM.java Locally Modified (Based On LOCAL)
@@ -1199,7 +1199,8 @@
          */
         public void doSvnRemoteLocationCheck(final StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
             // this can be used to hit any accessible URL, so limit that to admins
-            new FormFieldValidator(req, rsp, true) {
+            // no need to check for admin i suppose
+            new FormFieldValidator(req, rsp, false) {
                 protected void check() throws IOException, ServletException {
                     // syntax check first
                     String url = Util.nullify(request.getParameter("value"));
Index: src/main/java/hudson/tasks/BuildTrigger.java
--- src/main/java/hudson/tasks/BuildTrigger.java Base (BASE)
+++ src/main/java/hudson/tasks/BuildTrigger.java Locally Modified (Based On LOCAL)
@@ -272,7 +272,8 @@
          * Form validation method.
          */
         public void doCheck( StaplerRequest req, StaplerResponse rsp ) throws IOException, ServletException {
-            new FormFieldValidator(req,rsp,true) {
+            //no need to check for admin here
+            new FormFieldValidator(req,rsp,false) {
                 protected void check() throws IOException, ServletException {
                     String list = request.getParameter("value");
 
Index: src/main/java/hudson/triggers/TimerTrigger.java
--- src/main/java/hudson/triggers/TimerTrigger.java Base (BASE)
+++ src/main/java/hudson/triggers/TimerTrigger.java Locally Modified (Based On LOCAL)
@@ -59,7 +59,8 @@
          * Performs syntax check.
          */
         public void doCheck(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
-            new FormFieldValidator(req,rsp,true) {
+            //no need to check dor admin here
+            new FormFieldValidator(req,rsp,false) {
                 protected void check() throws IOException, ServletException {
                     try {
                         String msg = CronTabList.create(fixNull(request.getParameter("value"))).checkSanity();
Index: src/main/java/hudson/util/FormFieldValidator.java
--- src/main/java/hudson/util/FormFieldValidator.java Base (BASE)
+++ src/main/java/hudson/util/FormFieldValidator.java Locally Modified (Based On LOCAL)
@@ -160,7 +160,9 @@
         public URLCheck(StaplerRequest request, StaplerResponse response) {
             // can be used to check the existence of any file in file system
             // or other HTTP URLs inside firewall, so limit this to the admin.
-            super(request, response, true);
+            //This is used only in Sventon.java, FishEyeCVS.java and FishEyeSVN.java
+            //no need to check for admin
+            super(request, response, false);
         }
 
         /**
